### PR TITLE
Migration to separate devtools exchange from devtools extension

### PR DESCRIPTION
**Pull request by andyrichardson**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "@types/styled-components": "^4.1.18",
     "@typescript-eslint/eslint-plugin": "^1.11.0",
     "@typescript-eslint/parser": "^1.11.0",
+    "@urql/devtools": "../urql-devtools-exchange/",
```
Comment (JoviDeCroock) [inline package.json]: Can we use the name from the npm tarball? We can yarn link if needed

File: src/background/EventBus.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export class BackgroundEventTarget<T extends any = any> {
+  private listeners: Record<string, Handler<T> | undefined> = {};
+
+  public addEventListener(source: string, callback: Handler<T>) {
+    console.log("adding lsitenet", source);
```
Comment (JoviDeCroock) [inline src/background/EventBus.ts]: Typo "listener"

File: src/content_script/index.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import {
+  DevtoolsExchangeOutgoingMessage,
+  DevtoolsExchangeOutgoingEventType,
+  DevtoolsExchangeIncomingEventType
+} from "@urql/devtools/dist/index";
```
Comment (JoviDeCroock) [inline src/content_script/index.ts]: is this explicit dist ref needed?

File: package.json (hunk 1 above)
Comment (andyrichardson) [reply package.json]: This needs to be changed but won't work until a new devtools package is released with the types.

```suggestion
    "@urql/devtools": "~0.0.3",
```

File: src/background/EventBus.ts (hunk 2 above)
Comment (andyrichardson) [reply src/background/EventBus.ts]: What the hell happened here... I must have been drinking?

File: package.json (hunk 1 above)
Comment (JoviDeCroock) [reply package.json]: Let's make work of that release, I like the split up a lot

File: src/panel/context/Events.tsx [hunk 4]
```diff
@@ -62,16 +65,16 @@ export const EventsProvider: FC = ({ children }) => {
...
-  useEffect(() => {
-    window.chrome.devtools.inspectedWindow.eval(
-      `window.__urql__.events`,
-      (ops: UrqlEvent[]) => {
-        console.log(ops);
-        setRawEvents(ops);
-      }
-    );
-  }, []);
+  // /** Set initial state from cache */
```
Comment (JoviDeCroock) [inline src/panel/context/Events.tsx]: Does this happen automatically now or are we still missing this?

File: src/types/connections.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const ContentScriptConnectionName = "urql-cscript" as const;
```
[the remark this answers is not in the document]
Comment (andyrichardson) [reply src/types/connections.ts]: String literal types - so the line is a little more blurred.

File: src/background/EventBus.ts (hunk 2 above)
Comment (andyrichardson) [reply src/background/EventBus.ts]:
```suggestion
    console.log("adding listener", source);
```

Comment (andyrichardson) [inline src/background/EventBus.ts]:
```suggestion
    console.log("adding listener", source);
```

File: src/types/connections.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (JoviDeCroock) [reply src/types/connections.ts]: All right, well I guess if we consider this as a convention within our projects I can accept it :D I see these more as constants but no need to change it if we'll do it like this

File: src/panel/context/Events.tsx (hunk 4 above)
Comment (andyrichardson) [reply src/panel/context/Events.tsx]: We are missing this, my thinking is we should implement the panel similar to how chrome does the network panel. Once the panel is initialy opened, requests are captured.

**Pull request merged**